plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/move/MoveTypeChange.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Mario Kastner
+ *     - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+package org.eclipse.fordiac.ide.typemanagement.refactoring.move;
+
+import java.util.Optional;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.fordiac.ide.model.IdentifierVerifier;
+import org.eclipse.fordiac.ide.model.commands.change.ChangePackageNameCommand;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibrary;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.AbstractCommandChange;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.IFordiacModifiablePreviewChange;
+import org.eclipse.fordiac.ide.typemanagement.wizards.Messages;
+import org.eclipse.gef.commands.Command;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+public class MoveTypeChange extends AbstractCommandChange<LibraryElement> implements IFordiacModifiablePreviewChange {
+
+	private String newPackageName;
+	private String oldPackageName;
+	private final TypeEntry type;
+	private final URI destination;
+
+	protected MoveTypeChange(final String newPackageName, final TypeEntry type, final String name, final URI uri,
+			final Class<LibraryElement> elementClass, final URI newURI) {
+		super(name, uri, elementClass);
+		this.newPackageName = newPackageName;
+		this.type = type;
+		this.destination = newURI;
+	}
+
+	@Override
+	public void initializeValidationData(final LibraryElement element, final IProgressMonitor pm) {
+		oldPackageName = type.getPackageName();
```

Comment:
Please initialize the validation data from the passed `element`:
```suggestion
		oldPackageName = PackageNameHelper.getPackageName(element);
```